dot/core/service_test.go

```diff
@@ -28,6 +36,68 @@ import (
 
 var errTestDummyError = errors.New("test dummy error")
 
+func generateExtrinsic(t *testing.T) (ext types.Extrinsic, externExt types.Extrinsic, body *types.Body) {
+	rawMeta := common.MustHexToBytes(testdata.NewTestMetadata())
+	var decoded []byte
+	err := scale.Unmarshal(rawMeta, &decoded)
+	require.NoError(t, err)
+
+	meta := &ctypes.Metadata{}
+	err = ctypes.DecodeFromBytes(decoded, meta)
+	require.NoError(t, err)
+
+	testAPIItem := runtime.APIItem{
+		Name: [8]byte{1, 2, 3, 4, 5, 6, 7, 8},
+		Ver:  99,
+	}
+	rv := runtime.NewVersionData(
+		[]byte("polkadot"),
+		[]byte("parity-polkadot"),
+		0,
+		25,
+		0,
+		[]runtime.APIItem{testAPIItem},
+		5,
+	)
+	require.NoError(t, err)
+
+	bob, err := ctypes.NewMultiAddressFromHexAccountID(
+		"0x90b5ab205c6974c9ea841be688864633dc9ca8a357843eeacf2314649965fe22")
+	require.NoError(t, err)
+
+	call, err := ctypes.NewCall(meta, "Balances.transfer", bob, ctypes.NewUCompactFromUInt(12345))
+	require.NoError(t, err)
+
+	// Create the extrinsic
+	extrinsic := ctypes.NewExtrinsic(call)
+	genHash, err := ctypes.NewHashFromHexString("0x35a28a7dbaf0ba07d1485b0f3da7757e3880509edc8c31d0850cb6dd6219361d")
```

Comment:
also where is this from

Reply:
hmm I got it from here https://github.com/ChainSafe/gossamer/blob/4617175169f6e475764a553de653f34e6977ba9f/lib/babe/build_integration_test.go#L283 but not sure where that was from :/